mat64/cholesky.go

```diff
@@ -178,3 +178,154 @@ func (s *SymDense) InverseCholesky(chol *Cholesky) error {
 	s.SymOuterK(1, &t)
 	return err
 }
+
+// RankOne performs a rank-1 update of the Cholesky factorization orig as if the
+// update had been applied to the original matrix A, storing the result into the
+// reciever. That is, if in the original Cholesky factorization
+//  U^T * U = A,
+// in the updated factorization
+//  U'^T * U' = A + alpha * x * x^T = A'.
+//
+// Note that when alpha is negative, the updating problem may be ill-conditioned
+// and the results may be inaccurate, or the updated matrix A' may not be
+// positive definite and not have a Cholesky factorization. RankOne returns
+// whether the updated matrix A' is positive definite.
+//
+// The utility of RankOne comes from the fact that updating the factorization is
+// an O(n²) algorithm while computing it from scratch is an O(n³) algorithm.
+func (c *Cholesky) RankOne(orig *Cholesky, alpha float64, x *Vector) (ok bool) {
+	n := orig.Size()
+	if x.Len() != n {
+		panic(matrix.ErrShape)
+	}
+	if orig != c {
+		if c.chol == nil {
```

Comment:
I know this isn't done in LU (it probably should be), but could you make an isZero() function? It's a lot clearer code-wise to see it that way.